packages/suite/src/dsl.ts

```diff
@@ -35,31 +66,38 @@ export class TestBuilder<C extends Context> implements TestImplementation {
     this.children = test.children;
   }
 
-  step<R extends Context | void>(step: StepDefinition<C,R>): TestBuilder<R extends void ? C : C & R>;
-  step<R extends Context | void>(description: string, action: Action<C,R>): TestBuilder<R extends void ? C : C & R>;
```

Comment:
The value, imho in having these overloads is so that the different ways of invoking the function are apparent from intellisense in a way that wouldn't be true if it's just `...args: StepArgs<C,R>`. Can we refactor to have the base method to be `...args: StepArgs` and also keep the overloads?